sdlib/d/gc/emap.d

```diff
@@ -38,13 +37,25 @@ public:
 		return remap(extent, ExtentClass.large());
 	}
 
+	bool extend(Extent* extent, size_t size) shared {
+		assert(size > extent.size, "Size must be larger than current size!");
+
+		return map(extent.address + extent.size, size - extent.size,
+		           PageDescriptor(extent, extent.extentClass));
+	}
```

Comment:
This doesn't seem appropriate, the extent size must be updated BEFORE the emap is updated.